book-content/chapters/11-annotations-and-assertions.md

```diff
@@ -0,0 +1,1196 @@
+
+### Annotating Values With `satisfies`
+
+The `satisfies` operator is a way to tell TypeScript that a value must satisfy certain criteria, but still allow TypeScript to infer the type.
+
+Let's use it to make sure our `config` object has the right shape:
+
+```typescript
+const config = {
+  foreground: { r: 255, g: 255, b: 255 },
+  background: { r: 0, g: 0, b: 0 },
+  border: "transparent",
+} satisfies Record<string, Color>;
+```
+
+Now, we get the best of both worlds. This means we can access the keys without any issues:
+
+```typescript
+config.foreground.r;
+
+config.border.toUpperCase();
+```
+
+But we've also told TypeScript that `config` must be a `Record` with a string key and a `Color` value. If we try to add a key that doesn't match this shape, TypeScript will show an error:
+
+```ts twoslash
+// @errors: 2322
+type Color =
+  | string
+  | {
+      r: number;
+      g: number;
+      b: number;
+    };
+
+// ---cut---
+const config = {
+  primary: 123,
+} satisfies Record<string, Color>;
+```
+
+Of course, we have now lost the ability to add new keys to `config` without TypeScript complaining:
+
+```ts twoslash
+// @errors: 2339
+type Color =
+  | string
+  | {
+      r: number;
+      g: number;
+      b: number;
+    };
+
+const config = {} satisfies Record<string, Color>;
+// ---cut---
+config.somethingNew = "red";
+```
+
+Because TypeScript is now inferring `config` as _just_ an object with a fixed set of keys.
+
+Let's recap:
+
+- When you use a variable annotation, the variable's type wins.
+- When you don't use a variable annotation, the value's type wins.
+- When you use `satisfies`, you can tell TypeScript that a value must satisfy certain criteria, but still allow TypeScript to infer the type.
+
+#### Narrowing Values With `satisfies`
+
+A common misconception about `satisfies` is that it doesn't affect the type of the value. This is not quite true - in certain situations, `satisfies` does help narrow down a value to a certain type.
+
+Let's take this example:
+
+```tsx
+const album = {
+  format: "Vinyl",
+};
+```
+
+Here, we have an `album` object with a `format` key. As we know from our chapter on mutability, TypeScript will infer `album.format` as `string`. We want to make sure that the `format` is one of three values: `CD`, `Vinyl`, or `Digital`.
+
+We could give it a variable annotation:
+
+```tsx
+type Album = {
+  format: "CD" | "Vinyl" | "Digital";
+};
+
+const album: Album = {
+  format: "Vinyl",
+};
+```
+
+But now, `album.format` is `"CD" | "Vinyl" | "Digital"`. This might be a problem if we want to pass it to a function that only accepts `"Vinyl"`.
+
+Instead, we can use `satisfies`:
+
+```typescript
+const album = {
+  format: "Vinyl",
+} satisfies Album;
+```
+
+Now, `album.format` is inferred as `"Vinyl"`, because we've told TypeScript that `album` satisfies the `Album` type. So, `satisfies` is narrowing down the value of `album.format` to a specific type.
```

Comment:
more technical terms would be that the `satisfies` operator provides contextual information that helps the inferred value type to *retain* the literal string type here as otherwise it would be widened to a string. 

This is the same mechanism as with:
```ts
const album: Album = {
  format: "Vinyl",
};
```

Both of them provide the contextual type for that expression and it's what makes the literal string type to be retained in both of those examples

Reply:
What I'm perhaps picky about here is that I wouldn't call it narrowing here. TS speaks of narrowing in its control flow analysis - when it narrow a union type to one of its members in a branch etc